firmware/library/L1_Drivers/gpio.hpp

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <stdint.h>
+
+#include "L0_LowLevel/LPC40xx.h"
+
+class GpioInterface
+{
+ public:
+     virtual void SetAsInput(void)          = 0;
+     virtual void SetAsOutput(void)         = 0;
+     virtual void SetDir(bool input = true) = 0;
```

Comment:
Prefer that method and variable names be spelled out in full. This should be `SetDirection`.

Reply:
Also, looking at the input parameter, having a bool doesn't exactly read well. For example, `gpio.SetDirection(true);` doesn't immediately tell the user what is happening. True could mean output... or input, its a bit ambiguous. And setting it to default to true, means developers can do `gpio.SetDirection()`, but the behavior is still ambiguous.

It would be useful to create an `enum class` with constants such as, `kInput` and `kOutput`. See PinConfigInterface for an example: https://github.com/kammce/SJSU-Dev2/blob/master/firmware/library/L1_Drivers/pin_configure.hpp

Reply:
Would the Set method need a enum class as well, or is it intuitive enough to understand? (i.e. Set(true) means set high and Set(false) means set low)

Reply:
Also, should I just take out the SetDirection method?  If the user has to write a function that long that has restricted input in the form of enums, they could just write SetAsInput() or SetAsOutput(), which is shorter in written length.  The only benefit I see to having the SetDirection() method would be if it took in a boolean parameter, allowing users to put functions within the method in order to change the direction during runtime.  Even then, I don't think users would need to change the direction of the GPIO pin multiple times.  Any thoughts on that?

Reply:
I see your point. Alright, create a enum, and have the parameter be that enum type. Do not use a strongly typed "enum class", so that the user can use 0 (input) or 1 (output) as a means of changing directions.